test/libdnf/rpm/test_package_set.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "test_package_set.hpp"
+
+#include "libdnf/rpm/package.hpp"
+
+#include <cstdint>
+
+
+CPPUNIT_TEST_SUITE_REGISTRATION(RpmPackageSetTest);
+
+
+void RpmPackageSetTest::setUp() {
+    sack = new libdnf::rpm::Sack();
+
+    sack->pool = pool_create();
+    sack->pool->nsolvables = 32;
+
+    set1 = new libdnf::rpm::PackageSet(sack);
+    for (int i = 0; i < 16; i++) {
+        libdnf::rpm::Package pkg;
+        pkg.id = i;
+        set1->add(pkg);
+    }
+
+    set2 = new libdnf::rpm::PackageSet(sack);
+}
+
+
+void RpmPackageSetTest::tearDown() {
+    delete set1;
+    delete set2;
+    pool_free(sack->pool);
+    delete sack;
+}
+
+
+void RpmPackageSetTest::test_add() {
+    // add a Package that is out of range of PackageSet->pool->nsolvables
+    libdnf::rpm::Package pkg123;
+    pkg123.id = 123;
```

Comment:
This construct will be impossible in future. Package will have only a private constructor. Therefore I am not sure how to handle the test.

Reply:
That's right, I removed it as it's impossible.